File: device-sdk/src/main/java/io/oddworks/device/metric/OddVideoPlayMetric.java
package io.oddworks.device.metric;

public class OddVideoPlayMetric extends OddMetric {
    private static final String TAG = OddVideoPlayMetric.class.getSimpleName();

    protected static String videoPlayAction = ACTION_VIDEO_PLAY;
    protected static boolean videoPlayEnabled = false;

    public OddVideoPlayMetric() {

    }

    public static void setAction(String action) {
        videoPlayAction = action;
    }

    public static void setEnabled(boolean enabled) {
        videoPlayEnabled = enabled;
    }

    @Override
    public String getAction() {
        return videoPlayAction;
    }

    @Override
    public boolean getEnabled() {
        return videoPlayEnabled;
    }
}

File: device-sdk/src/main/java/io/oddworks/device/metric/OddVideoPlayMetric.kt
package io.oddworks.device.metric

import org.json.JSONObject

class OddVideoPlayMetric(contentType: String, contentId: String, title: String, meta: JSONObject? = null) : OddMetric(contentType, contentId, title, meta) {

    override val action: String
        get() = OddVideoPlayMetric.action

    override val enabled: Boolean
        get() = OddVideoPlayMetric.enabled

    companion object {
        private val TAG = OddVideoPlayMetric::class.java.simpleName

        var action = OddMetric.ACTION_VIDEO_PLAY
        var enabled = false
    }
}